From 596296f41cd7b3defb78499183e878cdfe3fe8e8 Mon Sep 17 00:00:00 2001
From: Aihua Xu <aihuaxu@apache.org>
Date: Wed, 24 Feb 2016 11:28:44 -0500
Subject: [PATCH 0814/1164] CDH-42565 HIVE-13129: CliService leaks HMS
 connection (Reviewed by Chaoyu Tang, Naveen
 Gangam)

(cherry picked from commit 2c570dece31968f8c4b6785461857098af1a742e)

Change-Id: Iae023814fdb6bba7440d1e9516399edfa397ecc3
---
 .../org/apache/hive/service/cli/CLIService.java    |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/service/src/java/org/apache/hive/service/cli/CLIService.java b/service/src/java/org/apache/hive/service/cli/CLIService.java
index f3bc109..978ff1d 100644
--- a/service/src/java/org/apache/hive/service/cli/CLIService.java
+++ b/service/src/java/org/apache/hive/service/cli/CLIService.java
@@ -147,6 +147,8 @@ public synchronized void start() {
   @Override
   public synchronized void stop() {
     super.stop();
+    // Release the HMS connection for this service thread
+    Hive.closeCurrent();
   }
 
   /**
-- 
1.7.9.5

